docs/extending/code-overview/apply-eval-format.rst

```diff
@@ -17,8 +17,8 @@ Also there is a list of all possible syntax styles styles which is not alterable
 in the ``Settings`` namespace. For Django, there is a Boolean setting indicating whether or
 not to use a Sans-Serif font.
 
-Evaluation
-----------
+Apply/Evaluate
```

Comment:
The blue box labeled 'Evaulation" in what has just been renamed to "apply-eval-format.png" should be changed to "Apply/Evaluate".

Reply:
I think that would be wrong. Evaluation implies applying replacement rules. There are two types of replacement rules:
* ``Rules`` (WL rules): a Pattern is associated with an S-Expression.
* ``BuiltinRules``: a Pattern is associated with a Python function (typically, some `apply_*` methods from a Builtin symbol class)

If the pattern has named subpatterns (leaves of the form `Pattern[symbolname_Symbol, subpat]`), in the case of WL rules, the corresponding symbols in the associated expression are replaced by the matching subexpressions. In the case of ``BuiltinRule``s, the expressions are passed to the apply method as named parameters.

Reply:
Here I am not talking about what Mathics calls "Evaluate". But "Apply/Eval" in the sense that is typically meant in functional programming, and defined in The Structure and Interpretation of Computer programs. 

So I think I'll change "Evaluate" to "Eval" (especially as it is often referred to as Appy/Eval). 

In Mathics, the `Expression.evaluate()` method is what in functional programming is called apply/eval .  It is probably too much work to change the name, and it isn't wrong because "evaluate" does sometimes mean "apply/eval"

Reply:
I just looked at the diagram. And here is how I think this should be changed


```
                            Evaluate
Input: str  -> S-Expression  -> Expression
                                    |   Form
                                    V
				Expression


Python [green]
Apply/Eval  [blue]
```

The changes: 

* "WL evaluation (Applying rules)" becomes  "Apply/Eval" which is what we should start calling this to use the more common term. 
* "Evaluation" -> "Evaluate" .  We have a mismatch in the kind of words used Evaluation (noun) and Formatting (present tense verb). "Evaluate": is the WL function that we use for apply/eval.  Although there isn't a WL function called "Form", there are many "xxxForm" functions and that is what we mean here. The boxing process is a special case of "Evaluate". And that's why the front-end should have been directing that rather than hiding this from the front end, registering acceptable forms and letting core decide. 
* "S-Expression" -> "Expression". Except for the initial parse, Expression can contain a lot of other information that is added as a result of Evaluate.